src/entities/asset/AssetIcon.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <div class="asset">
+    <div class="asset__icon">
+      <XOR v-if="nameLower=='xor'" />
+      <VAL v-else-if="nameLower=='val'" />
+      <ETH v-else-if="nameLower=='eth'" />
+      <IrohaIcon v-else-if="nameLower=='iroha'" />
+      <Soshiba v-else-if="nameLower=='soshiba'" />
+      <Pswap v-else-if="nameLower=='pswap'" />
+      <NaIcon v-else />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { toRefs } from 'vue';
+
+import XOR from '~icons/assets/XOR.svg';
+import VAL from '~icons/assets/VAL.svg';
+import ETH from '~icons/assets/ETH.svg';
+import IrohaIcon from '~icons/assets/iroha_icon.svg';
+import Soshiba from '~icons/assets/soshiba.svg';
+import Pswap from '~icons/assets/pswap.svg';
+import NaIcon from '~icons/assets/na.svg';
+
+const props = defineProps({
+  name: {
+    type: String,
+    default: '',
+  },
+});
+
+const { name } = toRefs(props);
+
+let nameLower = '';
+
+const init = async () => {
+  nameLower = name.value.toLowerCase();
+};
+
+init();
```

Comment:
That is not how we write Vue applications )

That's how:

```ts
import { computed } from 'vue'

const nameLower = computed(() => props.name.toLowerCase())
```

Please read the [Computed Properties](https://vuejs.org/guide/essentials/computed.html) section (with enabled "Composition" toggle).